src/main/kotlin/com/exactpro/th2/codec/AbstractCodecProcessor.kt

```diff
@@ -124,6 +128,23 @@ abstract class AbstractCodecProcessor<BATCH, GROUP, MESSAGE>(
             return null
         }
 
+        var mismatchDetested = false
+        messageGroup.groupItems.forEach { message ->
+            val messageBook: String = message.book(batch)
+            val eventBook: String? = message.eventBook
+            if (eventBook?.equals(messageBook) == false) {
+                eventProcessor.onErrorEvent(
+                    "Book name mismatch in '$messageBook' message and '$eventBook' parent event ids",
+                    message.eventId,
+                    listOf(message.id(batch))
+                )
+                mismatchDetested = true
+            }
+        }
+        if (mismatchDetested) {
+            return null
+        }
```

Comment:
Why do we drop the group if it has some issues with books in message ID and parent event ID? I don't think that should affect the codec work in any way...

Reply:
I thought that different book in message and event is critical problem, but I think is usual case for example for data-provider. Also, we can solve the problem via message reference

Reply:
> I thought that different book in message and event is critical problem

It is a critical problem only if this message is referenced in this event with a different book but here it is just a parent event ID attached to the message. I don't think we should drop the whole decoding/encoding result in this case